PresentData/PresentMonTraceConsumer.cpp

```diff
@@ -1139,6 +1188,169 @@ void PMTraceConsumer::HandleDWMEvent(EVENT_RECORD* pEventRecord)
     }
 }
 
+void PMTraceConsumer::RemovePresentFromTemporaryTrackingCollections(std::shared_ptr<PresentEvent> p)
+{
+    // Remove the present from any struct that would only host the event temporarily.
+    // Currently defined as all structures except for mPresentsByProcess, 
+    // mPresentsByProcessAndSwapChain, and mAllPresents.
+
+    // mPresentByThreadId
+    {
+        auto eventIter = mPresentByThreadId.find(p->ThreadId);
+        if (eventIter != mPresentByThreadId.end() && eventIter->second == p) {
+            mPresentByThreadId.erase(eventIter);
+        }
+        // TODO: Assert that an element has been removed if this element must exist.
+    }
+
+    // mPresentsBySubmitSequence
+    {
+        // TODO: is 0 a valid QueueSubmitSequence?
+        if (p->QueueSubmitSequence != 0) {
+            auto eventIter = mPresentsBySubmitSequence.find(p->QueueSubmitSequence);
+            if (eventIter != mPresentsBySubmitSequence.end() && (eventIter->second == p)) {
+                mPresentsBySubmitSequence.erase(eventIter);
+                // TODO: Assert elementsRemoved = 1 if this element must exist.
+            }
+        }
+    }
+
+    // mWin32KPresentHistoryTokens
+    {
+        if (p->ValidHistoryTokenKey) {

```

Comment:
You can just delete this and go based on the CompositionSurfaceLuid being nonzero.